sycl/doc/extensions/deprecated/sycl_ext_intel_online_compiler.asciidoc

```diff
@@ -0,0 +1,203 @@
+= SYCL Intel extension: Online Compilation
+Konstantin Bobrovskii <konstantin.s.bobrovsky@intel.com>, John Pennycook <john.pennycook@intel.com>
+v0.1
+:source-highlighter: pygments
+:icons: font
+== Introduction
+This document describes an interface for online compilation from high-level languages, such as
+OpenCL, to a binary format, such as SPIR-V, loadable by SYCL backends. Unlike SYCL 2020 provisional's
+OpenCL backend online compilation interface, this interface is not bound to any particular backend and does
+not require available SYCL context for online compilation.
+
+This gives a flexibility to "cross-compile" to SPIR-V or other supported formats without any SYCL
+device or context available. The online compilation API uses the `online_compiler` class to access
+compilation services. Instances of the class are constructed based on a specification of the desired
+compilation target passed to the constructors - such as compiled code format, target architecture,
+etc. All the settings are optional, and by default the target is generic SPIR-V.
+
+This API is an Intel SYCL extension.
+
+== Status
+
+This extension has been deprecated. Although it is still supported in DPC++, we expect that the interfaces defined in this specification will be removed in an upcoming DPC++ release. Shipping software products should stop using APIs defined in this specification and use an alternative instead.
```

Comment:
Just a minor nit.  Please respect the 80-column limit, and include the bold font that we have in the [template](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc):

```
This extension has been deprecated.  Although it is still supported in {dpcpp},
we expect that the interfaces defined in this specification will be removed in
an upcoming {dpcpp} release.  *Shipping software products should stop using
APIs defined in this specification and use an alternative instead.*
```

You will also need to define this near the top of the file (again see the template):

```
:dpcpp: pass:[DPC++]
```

Reply:
updated in https://github.com/intel/llvm/pull/16575/commits/9891f9d423ce8732931bfc7d4bddacd1a030a70f